docs/v1/P4Runtime-Spec.mdk

```diff
@@ -2396,27 +2400,61 @@ the following fields:
       found.
 
     * `translated_type`, if and only if the P4 `type` declaration was annotated
-      with `@p4runtime_translation`. It is of type `P4NewTypeTranslation`, which
-      itself has two fields --- `uri` and `sdn_bitwidth` ---, which map to the
-      two input parameters to the annotation.
+      with `@p4runtime_translation`. It is of type `P4NewTypeTranslation`,
+      which itself has two fields --- `uri` and either `sdn_string` or
+      `sdn_bitwidth` ---, which map to the two input parameters to the
+      annotation.
 
 * `annotations`, a repeated field of strings, each one representing a P4
   annotation associated to the type declaration.
 
 For example, an architecture --- in this case PSA --- may introduce a new type
 for port numbers:
 ~ Begin P4Example
+// Controller refers to ports as a string, e.g. "eth0".
+@p4runtime_translation("p4.org/psa/v1/PortId_t", string)
+type bit<9> PortId_String_t;
```

Comment:
This question need not cause this example to be changed, but to satisfy my own curiosity, if the control plane type is different, wouldn't you perhaps expect the URI to be different, too?  Having all of the 3 URIs in this example be identical to each other seems like something one would never want to do.  e.g. wouldn't you expect some code that performs the translation to use the URI to decide how to do it, and perhaps only the URI?

Reply:
done